Cabal/src/Distribution/PackageDescription/Check/Target.hs

```diff
@@ -294,6 +304,11 @@ checkBenchmark
           BenchmarkExeV10 _ f -> takeExtension (getSymbolicPath f) `notElem` [".hs", ".lhs"]
           _ -> False
 
+
+-- | Check if a module name is valid on both Windows and Posix systems
+checkModuleName :: Monad m => ModuleName -> CheckM m ()
+checkModuleName moduleName =
+  checkPackageFileNamesWithGlob PathKindFile (toFilePath moduleName)
```

Comment:
@ffaf1 I reused `checkPackageFileNamesWithGlob` here, as that also checked some things to do with tar.

I could implement a separate check with a separate warning category if you prefer that. Just let me know what is the best thing to do.

Reply:
The error seems clear enough, adding a specific function cna be done in the future if we need more granularity.